src/cmd/sign.js

```diff
@@ -107,6 +117,27 @@ export default function sign(
         downloadDir: artifactsDir,
       });
 
+      if (updateLink) {
+        let extensionID;
+        if (signingResult.id) {
+          extensionID = signingResult.id;
+        } else {
+          extensionID = id;
+        }
+        // flow thinks that oldUpdateManifestData can be potentially undefined
+        // this is wrong however, the if-case guarding both the `oldUpdateManifestData = `
+        // statement and this function call are both checking
+        // the exact same thing, which doesn't change
+        // anywhere in the code. Basically, `if (A) then define B` and later on `if (A) then use B`
+        // which flow doesn't seem to understand
+        let oldUpdateManifestDataMakeFlowHappy =
+          oldUpdateManifestData == null ? {} : oldUpdateManifestData;
```

Comment:
If you can push a change that shows the Flow error, I could take a look. I would rather not see extra variable assignments like this. I also can't spot a situation where it would be null but maybe it's because you didn't define the `Promise` return value from `fetchUpdateManifest`, I don't know.